replayer/replayer.cpp

```diff
@@ -102,6 +108,43 @@ std::istream& operator>>(std::istream& in, Replayer& o) {
   return in;
 }
 
+void Replayer::write(std::ostream& out) const {
+
+/*
+  flatbuffers::FlatBufferBuilder fbsBuilder;
+
+  std::vector<flatbuffers::Offset<fbs::Frame>> fbsFrames;
+  std::transform(
+    frames.begin(),
+    frames.end(),
+    fbsFrames.begin(),
+    [&fbsBuilder](const Frame& frame) { frame.addToFlatBufferBuilder(fbsBuilder); });
+
+  auto fbsFrameContainer = fbs::CreateFrameContainer(fbsBuilder, fbsBuilder.CreateVector(fbsFrames));
+  */
+}
+
+void Replayer::read(std::istream& out) {
+
+  validate();
+}
+
+void Replayer::validate() {
+  auto throwCorruptedReplayException = [&](const char* problem) {
+    std::string message("Corrupted replay: ");
+    message.append(problem);
+    std::runtime_error(message.c_str());
+  };
+
+  //TODO: Orders in a frame > 10000
+  //TODO: Players with actions in a frame > 9
+  //TODO: Actions in a frame > 10000
+  //TODO: Players with resources in a frame > 9
+  //TODO: Bullets in a frame > 500
+  //TODO: Players with units > 9
+  //TODO: Untis in a frame > 10000
```

Comment:
We used to do these kinds of assertions in the middle of the deserialization process. Some were intended to catch errors in the replay data; others were intended to catch serialization bugs, particularly to prevent issues with unexpected buffer sizes. FlatBuffers handles most of that now, so I intend to only retain the replay data assertions (ie. players <= 9, etc.)